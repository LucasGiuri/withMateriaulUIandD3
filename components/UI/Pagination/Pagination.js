import PropTypes from 'prop-types';
import { PaginationContainer, TitleContainer, ButtonContainer } from './Pagination.styles';
import Button from '../Button/Button';
import Title from '../Title/Title';

const Pagination = ({totalPages, hasPrevPage, hasNextPage, onClickPrev, onClickNext}) => (
  <PaginationContainer>
    <TitleContainer>
      <Title text={totalPages} />
    </TitleContainer>
    <ButtonContainer>
      <Button isDisabled={!hasPrevPage} onClick={onClickPrev} text={"Previous"} />
      <Button isDisabled={!hasNextPage} onClick={onClickNext} text={"Next"} />
    </ButtonContainer>
  </PaginationContainer>
);

Pagination.propTypes = {
  totalPages: PropTypes.string.isRequired,
  hasPrevPage: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
  hasNextPage: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
  onClickPrev: PropTypes.func,
  onClickNext: PropTypes.func
};

Pagination.defaultProps = {
  hasPrevPage: false,
  hasNextPage: false,
  onClickPrev: () => {},
  onClickNext: () => {}
};

export default Pagination;