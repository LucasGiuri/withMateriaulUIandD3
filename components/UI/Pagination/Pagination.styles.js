import styled from 'styled-components';
import { device } from '../../../public/styles/devices';

export const PaginationContainer = styled.div`
  padding-left: 14px;
  padding-right: 14px;
  
  @media ${device.mobile} {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
  }

  @media ${device.tablet} {
    flex-direction: row;
  }
`;

export const TitleContainer = styled.div`
  @media ${device.mobile} {
    width: 100%;
  }

  @media ${device.tablet} {
    width: auto;
  }
`;

export const ButtonContainer = styled.div`
  display:flex;
  justify-content: space-between;
  
  @media ${device.mobile} {
    width: 100%;
  }

  @media ${device.tablet} {
    width: auto;
  }
`;