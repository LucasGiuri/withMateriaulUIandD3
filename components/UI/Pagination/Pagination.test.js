import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';
import 'jest-styled-components';

describe('Pagination', () => {
  const totalPages = "Total Pages";
  const hasPrevPage = "http://google.com";
  const hasNextPage = "http://google.com";

  it('Pagination basic render snapshoot', () => {
    const component = renderer.create(<Pagination totalPages={totalPages} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Pagination hasNextPage render snapshoot', () => {
    const component = renderer.create(<Pagination totalPages={totalPages} hasNextPage={hasNextPage} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Pagination hasPrevPage render snapshoot', () => {
    const component = renderer.create(<Pagination totalPages={totalPages} hasPrevPage={hasPrevPage} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Pagination hasPrevPage and hasNextPage render snapshoot', () => {
    const component = renderer.create(<Pagination 
      totalPages={totalPages}
      hasPrevPage={hasPrevPage} 
      hasNextPage={hasNextPage} 
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Pagination onClickPrev is clicked', () => {
    const onClickPrev = jest.fn();
    const component = (<Pagination totalPages={totalPages} hasPrevPage={hasPrevPage} onClickPrev={onClickPrev} />);
    const tree = renderer.create(component);
    const childrenButtons = tree.root.findAllByType('button');
    childrenButtons.forEach(child => child.props.onClick())
    expect(onClickPrev).toHaveBeenCalledTimes(1);
  })

  it('Pagination onClickNext is clicked', () => {
    const onClickNext = jest.fn();
    const component = (<Pagination totalPages={totalPages} hasNextPage={hasPrevPage} onClickNext={onClickNext} />);
    const tree = renderer.create(component);
    const childrenButtons = tree.root.findAllByType('button');
    childrenButtons.forEach(child => child.props.onClick())
    expect(onClickNext).toHaveBeenCalledTimes(1);
  })
})
