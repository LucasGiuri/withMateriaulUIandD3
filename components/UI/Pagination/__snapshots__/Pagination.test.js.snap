// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Pagination basic render snapshoot 1`] = `
.c0 {
  padding-left: 14px;
  padding-right: 14px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 10px;
  color: #999999;
  cursor: default;
  border-color: #999999;
  background-color: #000000;
  font-size: 14px;
}

.c2 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: left;
  font-weight: normal;
}

@media (min-width:320px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:320px) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c1 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c3 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c3 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c4 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c4 {
    width: auto;
    margin-left: 14px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      Total Pages
    </p>
  </div>
  <div
    className="c3"
  >
    <button
      className="c4"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Pagination Pagination hasNextPage render snapshoot 1`] = `
.c0 {
  padding-left: 14px;
  padding-right: 14px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 10px;
  color: #999999;
  cursor: default;
  border-color: #999999;
  background-color: #000000;
  font-size: 14px;
}

.c5 {
  padding: 10px;
  color: #FFFFFF;
  cursor: pointer;
  border-color: #FFFFFF;
  background-color: #000000;
  font-size: 14px;
}

.c2 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: left;
  font-weight: normal;
}

@media (min-width:320px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:320px) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c1 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c3 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c3 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c4 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c4 {
    width: auto;
    margin-left: 14px;
  }
}

@media (min-width:320px) {
  .c5 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c5 {
    width: auto;
    margin-left: 14px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      Total Pages
    </p>
  </div>
  <div
    className="c3"
  >
    <button
      className="c4"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="c5"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Pagination Pagination hasPrevPage and hasNextPage render snapshoot 1`] = `
.c0 {
  padding-left: 14px;
  padding-right: 14px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c4 {
  padding: 10px;
  color: #FFFFFF;
  cursor: pointer;
  border-color: #FFFFFF;
  background-color: #000000;
  font-size: 14px;
}

.c2 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: left;
  font-weight: normal;
}

@media (min-width:320px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:320px) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c1 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c3 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c3 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c4 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c4 {
    width: auto;
    margin-left: 14px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      Total Pages
    </p>
  </div>
  <div
    className="c3"
  >
    <button
      className="c4"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="c4"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;

exports[`Pagination Pagination hasPrevPage render snapshoot 1`] = `
.c0 {
  padding-left: 14px;
  padding-right: 14px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  padding: 10px;
  color: #999999;
  cursor: default;
  border-color: #999999;
  background-color: #000000;
  font-size: 14px;
}

.c4 {
  padding: 10px;
  color: #FFFFFF;
  cursor: pointer;
  border-color: #FFFFFF;
  background-color: #000000;
  font-size: 14px;
}

.c2 {
  margin-bottom: 22px;
  padding-left: 14px;
  padding-right: 14px;
  font-family: 'Roboto',sans-serif;
  color: #DC3545;
  font-size: 22px;
  text-align: left;
  font-weight: normal;
}

@media (min-width:320px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:320px) {
  .c1 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c1 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c3 {
    width: 100%;
  }
}

@media (min-width:768px) {
  .c3 {
    width: auto;
  }
}

@media (min-width:320px) {
  .c5 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c5 {
    width: auto;
    margin-left: 14px;
  }
}

@media (min-width:320px) {
  .c4 {
    width: 49%;
  }
}

@media (min-width:768px) {
  .c4 {
    width: auto;
    margin-left: 14px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <p
      className="c2"
    >
      Total Pages
    </p>
  </div>
  <div
    className="c3"
  >
    <button
      className="c4"
      onClick={[Function]}
    >
      Previous
    </button>
    <button
      className="c5"
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
